### DISCUSSION TITLE
KeypointDetectorModel training time

### DISCUSSION THREAD

**Discussion by alenm10** [DonsetPG/narya#26] [Q&A] [answered]
Hi Paul,
Yesterday, I tried to train the whole KeypointDetectorModel from scratch, took mi over 4 hours in google colab and I got similar results. Inside that model you use efficientnetb3 for predicting keypoint masks. In your paper in Figure C.2 you showed results for other models (resnet50, inception,..) with all of them achieving between 0.81 and 0.89 IoU. I am curious about training time for other models if you can remember. Is any of those models that achieved lower accuracy any faster to train than efficientnet? Thank you

**Comment by DonsetPG** [comment, marked as answer]
Hey,

I just checked on my old colab (with a GPU): it takes me on average 53s per epoch. 
For the other models, the larger the number of parameters the longer to train the model will be as a rule of thumb. 

I think efficientnet models are pretty fast to train in comparison to the others. 

Hope this helps

**Reply by alenm10** [reply]
Thank you
